pkg/agent/flowexporter/connections/conntrack_connections_perf_test.go

```diff
@@ -0,0 +1,180 @@
+	conns = generateConns()
+	mockConnDumper.EXPECT().GetMaxConnections().Return(100000, nil).AnyTimes()
+
+	serviceStr := "10.0.0.1:30000/TCP"
+	servicePortName := k8sproxy.ServicePortName{
+		NamespacedName: types.NamespacedName{
+			Namespace: "serviceNS1",
+			Name:      "service1",
+		},
+		Port:     "30000",
+		Protocol: v1.ProtocolTCP,
+	}
+	mockProxier := proxytest.NewMockProxier(ctrl)
+	mockProxier.EXPECT().GetServiceByIP(serviceStr).Return(servicePortName, true).AnyTimes()
+
+	npQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+
+	connStore = NewConntrackConnectionStore(mockConnDumper, flowrecords.NewFlowRecords(), mockIfaceStore, true, false, mockProxier, npQuerier, testPollInterval)
+}
+
+func generateConns() []*flowexporter.Connection {
+	conns := make([]*flowexporter.Connection, testNumOfConns)
+	for i := 0; i < testNumOfConns; i++ {
+		conns[i] = getNewConn()
+	}
+	return conns
+}
+
+func generateUpdatedConns(conns []*flowexporter.Connection) []*flowexporter.Connection {
+	length := len(conns) - testNumOfDeletedConns + testNumOfNewConns
+	updatedConns := make([]*flowexporter.Connection, length)
+	for i := 0; i < len(conns); i++ {
+		// replace deleted connection with new connection
+		if conns[i].DoneExport == true {
+			conns[i] = getNewConn()
+		} else { // update rest of connections
+			conns[i].OriginalPackets += 5
+			conns[i].OriginalBytes += 20
+			conns[i].ReversePackets += 2
+			conns[i].ReverseBytes += 10
+		}
+		updatedConns[i] = conns[i]
+	}
+	for i := len(conns); i < length; i++ {
+		updatedConns[i] = getNewConn()
+	}
+	randomNum := getRandomNum(int64(length - testNumOfDeletedConns))
+	for i := randomNum; i < testNumOfDeletedConns+randomNum; i++ {
+		updatedConns[i].DoneExport = true
+	}
+	return updatedConns
+}
+
+func getNewConn() *flowexporter.Connection {
+	randomNum1 := getRandomNum(255)
+	randomNum2 := getRandomNum(255)
+	randomNum3 := getRandomNum(255)
+	src := net.ParseIP(fmt.Sprintf("192.%d.%d.%d", randomNum1, randomNum2, randomNum3))
+	dst := net.ParseIP(fmt.Sprintf("192.%d.%d.%d", randomNum3, randomNum2, randomNum1))
+	flowKey := flowexporter.Tuple{SourceAddress: src, DestinationAddress: dst, Protocol: 6, SourcePort: uint16(randomNum1), DestinationPort: uint16(randomNum2)}
+	return &flowexporter.Connection{
+		StartTime:                 time.Now().Add(-time.Duration(randomNum1) * time.Second),
+		StopTime:                  time.Now(),
+		IsPresent:                 true,
+		DoneExport:                false,
+		FlowKey:                   flowKey,
+		OriginalPackets:           10,
+		OriginalBytes:             100,
+		ReversePackets:            5,
+		ReverseBytes:              50,
+		SourcePodNamespace:        "ns1",
+		SourcePodName:             "pod1",
+		DestinationPodNamespace:   "ns2",
+		DestinationPodName:        "pod2",
+		DestinationServiceAddress: net.ParseIP("10.0.0.1"),
+		DestinationServicePort:    30000,
+		TCPState:                  "SYN_SENT",
```

Comment:
this is a bit misleading IMO. I thought this information was populated by `Poll` itself, it should not be populated in the return value of `DumpFlows`.

Reply:
Yes you are right. Updated.